clang/runtime/dpct-rt/include/dpct/math.hpp

```diff
@@ -2216,6 +2218,143 @@ void ldmatrix(uintptr_t addr, T *m1, T *m2, T *m3, T *m4, bool trans = false) {
   ldmatrix(addr, m4, trans, 3);
 }
 
+/// Multiplies 2 matrices (A & B) and adds the result to C matrix and
+/// accumulates the result to a D matrix (MAD). Requires the sub-group size of
+/// kernel calling this function to be 32.
+/// Current supported shapes & types:
+/// - m16n8k16 (f32.f16.f16.f32 & s32.s8.s8.s32)
+/// \tparam [in] M The rows of A, C & D matrix
+/// \tparam [in] N The columns of B, C, D matrix
+/// \tparam [in] K The columns & rows of A & B matrices respectively
+/// \tparam [in] MulType The type of A and B matrices in MMA ASM instruction
+/// \tparam [in] ABType The type of the input matrix (A & B) elements
+/// \tparam [in] CDType The type of the output matrix (C & D) elements
+/// \param [out] d_mat The elements of the output D matrix to store the result
+/// of A* B + C
+/// \param [in] a_mat The elements of the input A matrix to be multiplied with B
+/// matrix elements
+/// \param [in] b_mat The elements of the input B matrix to be multiplied with A
+/// matrix elements
+/// \param [in] c_mat The elements of the input C matrix to be added with the
+/// result of A * B
+template <int M, int N, int K, typename MulType, typename ABType,
```

Comment:
Can we merge MulType and ABType? The parameter type is void*, so the user's arguments can do an implicit cast to it.

Reply:
Done